vllm/v1/engine/core.py

```diff
@@ -775,7 +852,8 @@ def run_busy_loop(self):
                     logger.debug("Wave %d finished, pausing engine loop.",
                                  self.current_wave)
                     self.output_queue.put_nowait(
-                        EngineCoreOutputs(wave_complete=self.current_wave))
+                        (-1,
+                         EngineCoreOutputs(wave_complete=self.current_wave)))
```

Comment:
@njhill Why is this run when `self.local_dp_rank == 0`. Isn't it better to be `self.dp_rank == 0`? 
I understand you are resolving contention of multiple similar messages coming from local rank 0 of different nodes on the coordinator side. This means there is leakage from this logic to how coordinator is written.

Reply:
@kouroshHakha good catch, thanks! I think you're right that this should have been `self.dp_rank == 0`. However I don't think it will cause any problems if the coordinator gets this message from multiple ranks since it contains the wave number so it's just redundant.

Reply:
In our Ray PRs we may end up changing this due to the reason above. Good to know this lines up with our understanding. cc @ruisearch42